sdcm/nemesis.py

```diff
@@ -169,6 +169,22 @@ def disrupt_restart_node(self):
         self.log.info('Waiting JMX services to start after node restart')
         self.target_node.wait_jmx_up()
 
+    def disrupt_restart_with_resharding(self):
+        self._set_current_disruption('RestartNode with resharding %s' % self.target_node)
+        murmur3_partitioner_ignore_msb_bits = 15
+        self.log.info('Restart node with resharding. New murmur3_partitioner_ignore_msb_bits value: '
+                      '{murmur3_partitioner_ignore_msb_bits}'.format(**locals()))
+        self.target_node.restart_node_with_resharding(murmur3_partitioner_ignore_msb_bits=murmur3_partitioner_ignore_msb_bits)
+        self.log.info('Waiting scylla services to start after node restart')
+        self.target_node.wait_db_up()
+        self.log.info('Waiting JMX services to start after node restart')
+        self.target_node.wait_jmx_up()
+
+        # Wait 5 minutes our before return back the default value
+        time.sleep(360)
+        self.log.info('Set back murmur3_partitioner_ignore_msb_bits value to 12')
+        self.target_node.restart_node_with_resharding()
```

Comment:
How about wrapping the function statements to a try block? then the config will also be recovered when exception is raised.

Reply:
I am not sure I understand how can it helps here. Even it will be wrapped with the try block, it won't be recovered. We don't have roll back mechanism for failure in SCT

Reply:
But what is the aim of this? Even I revert the configuration and restart the cluster, the resharding won't be started. So the nemensis won't perform what it should

Reply:
Yes, it's probably something that should fail the longevity or at least get us an error event (when merging Larisa's events branch.